UCLALibrary/voyager-archive
# Add acquisitions-related database views and models

This PR relates to:
* [SYS-993](https://jira.library.ucla.edu/browse/SYS-993): Purchase orders
* [SYS-994](https://jira.library.ucla.edu/browse/SYS-994): Invoices
* [SYS-999](https://jira.library.ucla.edu/browse/SYS-999): Vendors

It adds multiple database views for these types of data.  For each there's a view for the "main" data, and one or more for related data.

Each new view has a corresponding Django model.  Two models have `TODO` notes, as they do not have natural single-column primary keys.  I designated a column in each model as primary key anyhow, but do not yet have sample data to test this.  If this causes a problem with real data, I'll derive a "natural" single-column PK or generate UUIDs or similar.

There's also a new function, used by views which need to convert "foreign" currency to US dollars.

The views / models probably still have inconsistencies, but the scope of this has grown too large.  These views work; they can be improved over time, especially when real data is available.

I did add some new sample data, but it's so tedious to find and pull rows like this that there still is not sample data for everything relevant.  Again, I hope to move to real data soon.

Testing: Not really needed, but to be thorough:
1) If running, shut down system: `docker-compose down`
2) Start system: `docker-compose up -d`
3) Migration 0003 should be applied automatically
4) Run: `sql_support/initialize_database.sh`
5) Start Django shell: `docker-compose exec django python manage.py shell`
6) Run these, you should see this, a queryset with 1 object:
```
>>> from voyager_archive.models import VendorView
>>> VendorView.objects.all()
<QuerySet [<VendorView: VendorView object (6718)>]>
```
